contextutil/detachcontext.go

```diff
@@ -0,0 +1,46 @@
+// Package contextutil adds functions for context.
+package contextutil // import "github.com/teamwork/utils/v2/contextutil"
+
+import (
+	"context"
+	"time"
+)
+
+// detachedContext is a context that is detached from the parent context.
+type detachedContext struct {
+	ctx context.Context
+}
+
+// Deadline returns the time when work done on behalf of this context
+// should be canceled. In a detached context, it will always return false.
+func (d detachedContext) Deadline() (time.Time, bool) {
+	return time.Time{}, false
+}
+
+// Done returns a nil channel always.
+func (d detachedContext) Done() <-chan struct{} {
+	return nil
+}
+
+// If Done is not yet closed, Err returns nil.
+// If Done is closed, Err returns a non-nil error explaining why:
+// Canceled if the context was canceled
+// or DeadlineExceeded if the context's deadline passed.
+// In a detached context, it will always return false.
```

Comment:
Ehh arguably the previous comment was more accurate considering we do always return nil on this detached context 😁

Reply:
okokokokookkokookokok done